src/plugins/plot/stackedPlot/mixins/objectStyles-mixin.js

```diff
@@ -128,36 +128,31 @@ export default {
       }
     },
     updateStyle(styleObj) {
-      let elemToStyle = this.getStyleReceiver();
+      const elemToStyle = this.getStyleReceiver();
 
-      if (!styleObj || elemToStyle === undefined) {
+      if (!styleObj || !elemToStyle) {
         return;
       }
 
-      let keys = Object.keys(styleObj);
-
-      keys.forEach((key) => {
-        if (elemToStyle) {
-          if (typeof styleObj[key] === 'string' && styleObj[key].indexOf('__no_value') > -1) {
-            if (elemToStyle.style[key]) {
-              elemToStyle.style[key] = '';
-            }
-          } else {
-            if (
-              !styleObj.isStyleInvisible &&
-              elemToStyle.classList.contains(STYLE_CONSTANTS.isStyleInvisible)
-            ) {
-              elemToStyle.classList.remove(STYLE_CONSTANTS.isStyleInvisible);
-            } else if (
-              styleObj.isStyleInvisible &&
-              !elemToStyle.classList.contains(styleObj.isStyleInvisible)
-            ) {
-              elemToStyle.classList.add(styleObj.isStyleInvisible);
-            }
-
-            elemToStyle.style[key] = styleObj[key];
+      // handle visibility separately
+      if (styleObj.isStyleInvisible !== undefined) {
+        elemToStyle.classList.toggle(STYLE_CONSTANTS.isStyleInvisible, styleObj.isStyleInvisible);
+        delete styleObj.isStyleInvisible;
+      }
+
+      // build style string
+      const styleString = Object.entries(styleObj)
+        .map(([key, value]) => {
+          if (typeof value === 'string' && value.includes('__no_value')) {
+            return `${key}: ;`; // removes the property
           }
-        }
+          return `${key}: ${value};`;
+        })
+        .join(' ');
+
+      // apply styles in one operation
+      requestAnimationFrame(() => {
+        elemToStyle.style.cssText += styleString;
       });
     }
   }
```

Comment:
Instead of using `cssText` to set this all in one go, we can set each property individually and wrap the whole thing in a raf call. There are only a few style properties so the loop shouldn't cause any significant delay

```suggestion
requestAnimationFrame(() => {
  Object.entries(styleObj).forEach(([key, value]) => {
    if (typeof value !== 'string' || !value.includes('__no_value')) {
      elemToStyle.style[key] = value;
    } else {
      elemToStyle.style[key] = ''; // remove the property
    }
  });
});
```

Reply:
yeah, I had that then over engineered it. This way is more performant for sure, I'll switch it up. Thanks!

Reply:
> yeah, I had that then over engineered it. This way is more performant for sure, I'll switch it up. Thanks!

All credit goes to my coworker Chad G. Petey